libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -104,17 +105,49 @@ bool ImmutableValidator::visit(MemberAccess const& _memberAccess)
 
 bool ImmutableValidator::visit(IfStatement const& _ifStatement)
 {
-	bool prevInBranch = m_inBranch;
+	auto prevInBranch = m_inBranch;
+	auto prevWrites = std::move(m_immutableStateVariablesWrittenInBranch);
```

Comment:
It seems dangerous to me to use `std::move()` here. You also access the moved variable a a few lines down and then move it again, yet to my understanding variables that have been moved should no longer be used...